From c37cc9b23048ac51229fd59643474db65c163bc5 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Mon, 26 May 2014 16:01:55 +0300
Subject: [PATCH 167/213] Version 0.9.3: Order of fields in Cow Registration
 Activity       2       3 The order of fields in cow registration has been
 changed to       4    - name       5    - ear tag number       6    - sex    
   7    - age       8    - Date of Birth       9    - Breed      10    -
 Deformity      11    - Others

---
 MistroFarmer/src/main/AndroidManifest.xml          |  4 +--
 .../mistro/farmer/AddMilkProductionActivity.java   |  2 +-
 .../mistro/farmer/CowRegistrationActivity.java     | 10 ++++--
 .../main/res/layout/activity_cow_registration.xml  | 40 ++++++++++++++--------
 MistroFarmer/src/main/res/values/dimens.xml        |  1 +
 MistroFarmer/src/main/res/values/strings_en.xml    | 29 ++++++++++++++--
 6 files changed, 64 insertions(+), 22 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 04d02d1..f7a055e 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
-    android:versionCode="2"
-    android:versionName="0.9.1" >
+    android:versionCode="3"
+    android:versionName="0.9.3" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
index 7e5216b..095b4be 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
@@ -363,7 +363,7 @@ public class AddMilkProductionActivity extends SherlockActivity implements Mistr
                                 Date currMilkingDate = new SimpleDateFormat(dateFormat, java.util.Locale.ENGLISH).parse(dateET.getText().toString());
                                 long currMilking = currMilkingDate.getTime();
 
-                                int dayDiff = (int)((currMilking - lastMilking)/86400000);
+                                int dayDiff = (int)Math.abs((currMilking - lastMilking)/86400000);
                                 int yieldDiff = Math.abs(Integer.parseInt(quantityET.getText().toString()) - lastMP.getQuantity());
 
                                 if(dayDiff != 0) {
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
index 9988fcf..fed2491 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
@@ -54,6 +54,7 @@ public class CowRegistrationActivity extends SherlockActivity implements MistroA
     public static final String KEY_INDEX="index";
     public static final String KEY_NUMBER_OF_COWS="numberOfCows";
     private final String dateFormat="dd/MM/yyyy";
+    private TextView nameETNHintTV;
     private TextView nameTV;
     private EditText nameET;
     private TextView earTagNumberTV;
@@ -125,6 +126,7 @@ public class CowRegistrationActivity extends SherlockActivity implements MistroA
         selectedBreeds = 0;
 
         //init views
+        nameETNHintTV = (TextView) this.findViewById(R.id.name_eta_hint_tv);
         nameTV=(TextView)this.findViewById(R.id.name_tv);
         nameET=(EditText)this.findViewById(R.id.name_et);
         earTagNumberTV=(TextView)this.findViewById(R.id.ear_tag_number_tv);
@@ -459,12 +461,14 @@ public class CowRegistrationActivity extends SherlockActivity implements MistroA
     public void initTextInViews() {
         String title = Locale.getStringInLocale("cow_registration",this)+" "+String.valueOf(index+1);
         this.setTitle(title);
+
+        nameETNHintTV.setText(" * " + Locale.getStringInLocale("ear_tag_no_or_name", this));
         nameTV.setText(Locale.getStringInLocale("name",this));
-        earTagNumberTV.setText(" * "+Locale.getStringInLocale("ear_tag_number",this));
+        earTagNumberTV.setText(Locale.getStringInLocale("ear_tag_number",this));
 
-        ageOrDOBHintTV.setText(Locale.getStringInLocale("age_or_dob", this));
+        ageOrDOBHintTV.setText(" * " + Locale.getStringInLocale("age_or_dob", this));
 
-        ageTV.setText(" * "+Locale.getStringInLocale("age",this));
+        ageTV.setText(Locale.getStringInLocale("age",this));
         int ageTypeArrayID = Locale.getArrayIDInLocale("age_type_array",this);
         if(ageTypeArrayID!=0){
             ArrayAdapter<CharSequence> ageTypeArrayAdapter=ArrayAdapter.createFromResource(this, ageTypeArrayID, android.R.layout.simple_spinner_item);
diff --git a/MistroFarmer/src/main/res/layout/activity_cow_registration.xml b/MistroFarmer/src/main/res/layout/activity_cow_registration.xml
index 64c2abc..5beda11 100644
--- a/MistroFarmer/src/main/res/layout/activity_cow_registration.xml
+++ b/MistroFarmer/src/main/res/layout/activity_cow_registration.xml
@@ -12,6 +12,15 @@
         android:paddingLeft="@dimen/activity_padding_side"
         android:paddingRight="@dimen/activity_padding_side"
         android:orientation="vertical">
+
+        <TextView
+            android:id="@+id/name_eta_hint_tv"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/activity_f_r_text_view_margin_top"
+            android:textSize="@dimen/small_text_size"
+            android:textStyle="italic" />
+
         <TextView
                 android:id="@+id/name_tv"
                 android:layout_width="wrap_content"
@@ -44,11 +53,26 @@
                 android:inputType="text"/>
 
         <TextView
+            android:id="@+id/sex_tv"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/activity_f_r_text_view_margin_top"
+            android:textSize="@dimen/normal_text_size"/>
+        <Spinner
+            android:id="@+id/sex_s"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/activity_f_r_edit_text_margin_top"
+            android:textColor="@color/text_input_color"
+            android:layout_marginLeft="@dimen/activity_f_r_edit_text_margin_left"/>
+
+        <TextView
             android:id="@+id/age_or_dob_hint_tv"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginTop="@dimen/activity_f_r_text_view_margin_top"
-            android:textSize="@dimen/normal_text_size" />
+            android:textSize="@dimen/small_text_size"
+            android:textStyle="italic" />
 
         <TextView
             android:id="@+id/age_tv"
@@ -113,19 +137,7 @@
             android:textSize="@dimen/normal_text_size"
             android:textColor="@color/text_input_color"
             android:inputType="date|none"/>
-        <TextView
-            android:id="@+id/sex_tv"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="@dimen/activity_f_r_text_view_margin_top"
-            android:textSize="@dimen/normal_text_size"/>
-        <Spinner
-            android:id="@+id/sex_s"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="@dimen/activity_f_r_edit_text_margin_top"
-            android:textColor="@color/text_input_color"
-            android:layout_marginLeft="@dimen/activity_f_r_edit_text_margin_left"/>
+
         <TextView
                 android:id="@+id/deformity_tv"
                 android:layout_width="wrap_content"
diff --git a/MistroFarmer/src/main/res/values/dimens.xml b/MistroFarmer/src/main/res/values/dimens.xml
index 35deb02..2893e74 100644
--- a/MistroFarmer/src/main/res/values/dimens.xml
+++ b/MistroFarmer/src/main/res/values/dimens.xml
@@ -5,6 +5,7 @@
     <dimen name="activity_landing_btn_layout_height">50dp</dimen>
     <dimen name="activity_landing_btn_side_margin">20dp</dimen>
     <dimen name="normal_text_size">20dp</dimen>
+    <dimen name="small_text_size">15dp</dimen>
     <dimen name="activity_padding_top">5dp</dimen>
     <dimen name="activity_padding_bottom">15dp</dimen>
     <dimen name="activity_padding_side">20dp</dimen>
diff --git a/MistroFarmer/src/main/res/values/strings_en.xml b/MistroFarmer/src/main/res/values/strings_en.xml
index 68f7c2d..13960a4 100644
--- a/MistroFarmer/src/main/res/values/strings_en.xml
+++ b/MistroFarmer/src/main/res/values/strings_en.xml
@@ -35,6 +35,30 @@
         <item>Months</item>
         <item>Years</item>
     </string-array>
+    <string-array name="c_breeds_array_en">
+        <item>Friesian</item>
+        <item>Jersey</item>
+        <item>Ankole</item>
+        <item>Boran</item>
+        <item>Ayrshire</item>
+        <item>Zebu</item>
+        <item>Sahiwal</item>
+        <item>Guernsey</item>
+        <item>Aberdeen Angus</item>
+        <item>Hereford</item>
+        <item>Nganda</item>
+    </string-array>
+    <string-array name="uc_breeds_array_en">
+        <item>Charolais</item>
+        <item>Simmental</item>
+        <item>Limousine</item>
+        <item>Brahman</item>
+        <item>Santa Getrude</item>
+        <item>Piemontese</item>
+        <item>Red Poll</item>
+        <item>Fleckviah</item>
+        <item>Brown Swiss</item>
+    </string-array>
     <string-array name="breeds_array_en">
         <item>Ayrshire</item>
         <item>Boran</item>
@@ -65,7 +89,7 @@
         <item>Clef palate</item>
         <item>Other</item>
     </string-array>
-    <string name="service_type_used_en">Service Type Used</string>
+    <string name="service_type_used_en">Service method Used</string>
     <string-array name="service_types_en">
         <item>Bull</item>
         <item>Artificial Insemination</item>
@@ -217,7 +241,7 @@
     <string name="enter_dam_etn_en">Enter dam\'s ear tag number</string>
     <string name="age_diff_from_dob_en">Cow\'s age appears to conflict with the cow\'s Date of birth</string>
     <string name="action_not_allowed_en">Action not allowed here</string>
-    <string name="are_you_in_farm_en">Are you in your farm?</string>
+    <string name="are_you_in_farm_en">Ng\'ombe Planner will try to record your farm\'s GPS coordinates. Are you currently in your farm?</string>
     <string name="farm_coords_successfully_reg_en">Farm coordinates successfully registered</string>
     <string name="sms_charges_en">SMS Charges</string>
     <string name="incur_network_sms_charges_en">You are not connected to the internet. You will therefore incur your standard network SMS charge to send data to the server.</string>
@@ -237,6 +261,7 @@
     <string name="you_do_not_have_cows_en">You do not seem to have cows</string>
 
     <!-- Added after translation-->
+    <string name="ear_tag_no_or_name_en">You can enter the either the cow\'s name or ear tag number</string>
     <string name="age_or_dob_en">You can enter either the cow\'s age or date of birth</string>
     <string name="enter_age_or_dob_en">Please enter either the cow\'s age or date of birth</string>
     <string name="cow_too_young_en">The selected cow is too young</string>
-- 
2.7.1

